playwright-tests/tests/infrastructure/rfp.spec.js

```diff
@@ -0,0 +1,309 @@
+
+    const submitbutton = await page.locator("button", { hasText: "submit" });
+    await submitbutton.scrollIntoViewIfNeeded();
+    await expect(submitbutton).toBeEnabled();
+    await pauseIfVideoRecording(page);
+    await submitbutton.click();
+
+    const transactionText = JSON.stringify(
+      JSON.parse(await page.locator("div.modal-body code").innerText()),
+      null,
+      1
+    );
+    await expect(transactionText).toEqual(
+      JSON.stringify(
+        {
+          labels: ["Explorers"],
+          body: {
+            rfp_body_version: "V0",
+            name: "test title",
+            description: "The RFP description",
+            summary: "the rfp summary",
+            submission_deadline: "1923264000000000000",
+            timeline: {
+              status: "ACCEPTING_SUBMISSIONS",
+            },
+          },
+        },
+        null,
+        1
+      )
+    );
+    await pauseIfVideoRecording(page);
+  });
+  test("should cancel RFP", async ({ page }) => {
+    await mockRpcRequest({
+      page,
+      filterParams: {
+        method_name: "get_rfp",
+      },
+      modifyOriginalResultFunction: async (originalResult) => {
+        console.log(JSON.stringify(originalResult, null, 1));
+        originalResult.snapshot.timeline.status = "ACCEPTING_SUBMISSIONS";
+        return originalResult;
+      },
+    });
+
+    await page.goto("/infrastructure-committee.near/widget/app?page=rfp&id=0");
+    const buttonSelector = `div[data-testid="setting-btn"]`;
+    await page.waitForSelector(buttonSelector, {
+      state: "visible",
+    });
+    await page.click(buttonSelector);
+    await page.getByRole("button", { name: "Accepting Submissions" }).click();
+    await page.getByText("Cancelled", { exact: true }).click();
+    await page.getByRole("radio").first().click();
+    await page.getByRole("button", { name: "Ready to Cancel" }).click();
+
+    const transactionText = JSON.stringify(
+      JSON.parse(await page.locator("div.modal-body code").innerText()),
+      null,
+      1
+    );
+    await expect(transactionText).toEqual(
+      JSON.stringify(
+        {
+          id: 0,
+          proposals_to_cancel: [],
+          proposals_to_unlink: [],
+        },
+        null,
+        1
+      )
+    );
+    await pauseIfVideoRecording(page);
+  });
+  test("should edit RFP", async ({ page }) => {
+    await mockRpcRequest({
+      page,
+      filterParams: {
+        method_name: "get_rfp",
+      },
+      modifyOriginalResultFunction: async (originalResult) => {
+        console.log(JSON.stringify(originalResult, null, 1));
+        originalResult.snapshot.timeline.status = "ACCEPTING_SUBMISSIONS";
+
+        return originalResult;
+      },
+    });
+
+    await page.goto("/infrastructure-committee.near/widget/app?page=rfp&id=0");
+    await page.getByRole("button", { name: "Edit" }).click();
+    await page.locator(".badge .bi-trash3-fill").click({ timeout: 1000 });
```

Comment:
This timeout of `1000` ms might be a bit to little. Just leave it to the default which is 5s.